kernel/src/log_segment.rs

```diff
@@ -162,4 +442,270 @@ mod tests {
         // missing columns, but can still skip part 3 because has valid nullcount stats for P&M.
         assert_eq!(data.len(), 4);
     }
+
+    fn get_path(index: usize, suffix: &str) -> Path {
```

Comment:
you can use the  brand new `delta_path_for_version` in test-utils. You can delete this method and replace all calls to it with calls to that.